src/components/ShaclForm/CustomDate.ts

```diff
@@ -0,0 +1,62 @@
+// eslint-disable-next-line no-shadow
+export enum XsdDateTypes {
+    // see FieldUtils.isDatetime()
+    DateTime = 'http://www.w3.org/2001/XMLSchema#dateTime',
+    Date = 'http://www.w3.org/2001/XMLSchema#date'
+    // No Vue Component that can only select year/month, so for now we disable these options.
+    // YearMonth = 'http://www.w3.org/2001/XMLSchema#gYearMonth',
+    // Year = 'http://www.w3.org/2001/XMLSchema#gYear'
+  }
+
+/*
+In rdf you have multiple date types (see XsdDataTypes). FDP code used the Typescript Date object
+but this is always serialized to format date/time.
+CustomDate is serialized to string based on the type. (xmls:gYear -> "2015")
+ */
+export class CustomDate {
+  private internalDate: Date
+
+  private readonly mode: XsdDateTypes
+
+  constructor(input: string | Date | number, xsdType: string) {
+    this.mode = xsdType as XsdDateTypes
+    this.internalDate = new Date(input)
+    this.normalize()
+  }
+
+  private normalize() {
+    switch (this.mode) {
+      // case XsdDateTypes.Year:
+      //   this.internalDate.setMonth(0) // fall through
+      // case XsdDateTypes.YearMonth:
+      //   this.internalDate.setDate(1) // fall through
+      case XsdDateTypes.Date:
+        this.internalDate.setHours(0, 0, 0, 0)
+        break
+      default:
+        // keep full date/time
+        break
+    }
+  }
+
+  // valueOf(): number {
+  //   return this.internalDate.valueOf()
+  // }
+
+  toString(): string {
+    const isoStr = moment(this.internalDate).toISOString(true)
```

Comment:
What does `moment` add here? Assuming `this.internalDate` is a native `Date` object (see L23), according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) it has a `toISOString()` method